### PR TITLE
Handle missing flow runs in k8s agent resource management

### PR DESCRIPTION
**Pull request by joshmeek**
## Summary
The resource management loop in the Kubernetes agent has the potential to error if it attempts to set the state for a missing flow run. This can happen in the case where a job is started, the user deletes the flow run, then the agent attempts to clean up the job (when the job is in a state where the agent marks the flow run as failed). To avoid this we handle the case where a `ClientError` is raised by the call to the client and the loop continues with its resource management cycle.

## Changes
Handles `ClientError`s when calling the `set_flow_run_state` mutation in the Kubernetes agent resource management loop

## Importance
Avoids a possible edge case

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/agent/kubernetes/agent.py [hunk 1]
```diff
@@ -320,14 +327,19 @@ def manage_jobs(self) -> None:
...
+                                    flow_run_id=flow_run_id,
+                                    state=Failed(
+                                        message="Kubernetes Error: pods {} failed for this job".format(
+                                            failed_pods
+                                        )
+                                    ),
+                                )
+                            except ClientError:
+                                self.logger.error(
+                                    f"Error attempting to set flow run state for {flow_run_id}"
```
Comment (madkinsz) [inline src/prefect/agent/kubernetes/agent.py:341]: Can you attach the `exc_info` ?

Comment (madkinsz) [reply src/prefect/agent/kubernetes/agent.py:341]: (and in the other place where this is logged)

**Review by madkinsz — APPROVED** [review]
